packages/core/utils/src/dml/properties/id.ts

```diff
@@ -1,28 +1,39 @@
 import { BaseProperty } from "./base"
+import { PrimaryKeyModifier } from "./primary-key"
+
+const IsIdProperty = Symbol("IsIdProperty")
 
 /**
  * The Id property defines a unique identifier for the schema.
  * Most of the times it will be the primary as well.
  */
 export class IdProperty extends BaseProperty<string> {
+  [IsIdProperty] = true
+
+  static isIdProperty(value: any): value is IdProperty {
+    return !!value?.[IsIdProperty]
+  }
+
   protected dataType: {
     name: "id"
     options: {
-      primaryKey: boolean
       prefix?: string
     }
   }
 
-  constructor(options?: { primaryKey?: boolean; prefix?: string }) {
+  constructor(options: { prefix?: string } = {}) {
     super()
+
     this.dataType = {
       name: "id",
-      options: { primaryKey: true, ...options },
+      options,
     }
   }
 
   primaryKey(decision: boolean) {
```

Comment:
question: completely unrelated, but just noticed: what does `decision` do?

Reply:
it might disapear but I believe it is used under the hood to reset the primary key to non primary key but it does not really make sense and instead use the PrimaryKeyModifier, but this pr is not yet ready and will include many modification that are incoming

Reply:
it has been resolved